src/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -232,13 +234,28 @@ def get_message_res(self, message_ids: set[UUID]) -> list[Message]:
         with self.lock:
             current = time.time()
 
-            # Verify Messge IDs
+            # Verify Message IDs
             ret = verify_message_ids(
                 inquired_message_ids=message_ids,
                 found_message_ins_dict=self.message_ins_store,
                 current_time=current,
             )
 
+            # Check node availability
+            dst_node_ids = {
+                self.message_ins_store[message_id].metadata.dst_node_id
+                for message_id in message_ids
+            }
+            tmp_ret_dict = check_node_availability_for_in_message(
+                inquired_in_message_ids=message_ids,
+                found_in_message_dict=self.message_ins_store,
+                node_id_to_online_until={
+                    node_id: self.node_ids[node_id][0] for node_id in dst_node_ids
+                },
+                current_time=current,
+            )
+            ret.update(tmp_ret_dict)
+
             # Find all reply Messages
             message_res_found: list[Message] = []
             for message_id in message_ids:
```

Comment:
@yan-gao-GY @panh99 do we need to handle a possible race condition here? What happens if we generate the error response, but a different thread inserts a reply message that the following lines of code would then include in the response?